OfficeDev/office-ui-fabric-react
# DatePicker: Improve screenreader accessibility

#### Pull request checklist

- [x] Addresses an existing issue: #1266
- [x] Include a change request file if publishing
- [ ] New feature, bugfix, or enhancement

#### Description of changes

The previous/next month/year buttons don't have aria-labels and therefore are indiscernible to screen reader users. This pull request fixes this by enabling consumers of the date-picker to provide aria-labels for the buttons.

This pull request also improves the navigation of the date picker for screen reader users by re-announcing the month and year whenever the user navigates across a month/year boundary. This behavior is adapted from some accessibility best practices laid out by the [OAA](http://www.oaa-accessibility.org/examplep/datepicker1/).

#### Focus areas to test

Test steps (verified manually on DatePicker.Basic.Example):
1) Open test site in Edge, turn on Narrator, navigate to the date picker via keyboard, and open the date picker.
2) Navigate from the last day of a month to the first day of the next month. The new month should be read out by the screen reader.
3) Navigate to the previous/next month/year buttons. The screen reader should read out "Go to {previous,next} {month,year}"
